D7 LTS Usage and Requirements (#9290)
pantheon-systems/documentation: source/content/supported-drupal.md

Also included, is continued support for Drupal 7-compatible runtime environments on the Pantheon Platform, including PHP, MySQL and other prerequisites.

##### Requirements
* Available to all site plans and workspace tiers.
* Sites must use the [Pantheon Drupal 7 Upstream](https://github.com/pantheon-systems/drops-7).
  * Or a Custom Upstream based on the Pantheon Drupal 7 Upstream with the [latest updates pulled in](/guides/custom-upstream/create-custom-upstream#pull-in-core-from-pantheons-upstream).
* Sites must be updated to the latest release of Drupal core (currently 7.101).
<br/>
<br/>

To verify your current Drupal core version, you can check the [Status report tab](/drupal-launch-check#status-report) in the Pantheon site dashboard or you can use Terminus: 

```bash{promptUser: user}
terminus drush <site>.<env> status
```

To verify your current upstream, you can go to the Pantheon site dashboard and click **Settings**, then **About Site** or you can use Terminus: 

```bash{promptUser: user}
terminus site:info <site>
```

> If you are not already using the Pantheon Drupal 7 upstream, you can [switch the site upstream via Terminus](/terminus/examples#switch-upstreams).

Reviewer: If folks run this command without ensuring that their repository is aligned with the drops-7 repository, then the next dashboard update command could result in merge conflicts (best case) or destructive changes to the site.

* (Recommended) Not a requirement, but we strongly recommend updating all contrib modules to their latest versions as well. 

##### D7ES Usage
By enabling this module, your site’s list of contrib modules will be automatically submitted for D7ES security coverage. You will receive email notifications of D7 core & installed contrib module security and compatibility updates. 

Enable the module, then go to `/admin/config/system/tag1-d7es` and configure the list of email addresses that should be notified when applicable updates are released. On this configuration page you can also confirm your sites connection with Tag1’s update servers.

Apply contrib updates from the Tag1 D7ES service via Drush, Autopilot, or manually via SFTP. For details, see the following tabs.

<TabList>

<Tab title="Drush (Recommended)" id="drush" active={true}>

1. [Set the connection mode to SFTP](/guides/sftp) for the Dev or Multidev environment via the Pantheon Dashboard or with [Terminus](/terminus):

 ```bash{promptUser: user}
 terminus connection:set <site>.<env> sftp
 ```

1. Use Drush to apply contrib updates from the Tag1 D7ES service: 
 ```bash{promptUser: user}

>  terminus connection:set <site>.<env> pm:update --no-core

Author: @scottbuscemi can you confirm whether `--no-core` is needed here? The gdoc did not have this flag but I think it's needed so that our upstream core files aren't overwritten

Reviewer: Yes. This file should prevent core from being overwritten:

https://github.com/pantheon-systems/drops-7/blob/default/.drush-lock-update

I don't recall whether you get an error or warning if `--no-core` is omitted, but it's best to include it for clarity.

Refer to [Create a New CMS Site](/add-site-dashboard) for how to create a new Drupal 7 site from the Pantheon dashboard.

### Drupal 7 long-term support

> Drupal 7 will reach it's end of life on January 5, 2025. Pantheon has partnered with Tag1 Consulting to offer Long-Term Support for Drupal 7 through **January 5, 2027**.

Reviewer 2: ```suggestion
Drupal 7 will reach it's end of life on January 5, 2025. Pantheon has partnered with [Tag1 Consulting](https://www.tag1consulting.com/) to offer Long-Term Support for Drupal 7 through **January 5, 2027**.
```

> To learn more about this partnership, see [related blog post](https://pantheon.io/blog/pantheon-and-tag1-support-drupal-7-websites).

Reviewer 2: ```suggestion
To learn more about this partnership, see the [related blog post](https://pantheon.io/blog/pantheon-and-tag1-support-drupal-7-websites).
```

#### D7 Core LTS and Platform Compatibility 

> This includes security and compatibility updates to Drupal core and core dependencies from Tag1 Consulting, which are distributed by the Pantheon Drupal 7 Upstream and made available as a one-click updates via [the Pantheon Dashboard](core-updates#apply-upstream-updates-via-the-site-dashboard) or [Terminus command line tool](/core-updates#apply-upstream-updates-via-terminus).

Reviewer 2: ```suggestion
The D7 Long-Term Support includes security and compatibility updates to Drupal core and core dependencies from Tag1 Consulting, which are distributed by the Pantheon Drupal 7 Upstream and made available as a one-click updates via [the Pantheon Dashboard](core-updates#apply-upstream-updates-via-the-site-dashboard) or [Terminus command line tool](/core-updates#apply-upstream-updates-via-terminus).
```

> Also included, is continued support for Drupal 7-compatible runtime environments on the Pantheon Platform, including PHP, MySQL and other prerequisites.

Reviewer 2: ```suggestion
Also included is continued support for Drupal 7-compatible runtime environments on the Pantheon Platform, including PHP, MySQL and other prerequisites.
```

##### Requirements

> * Available to all site plans and workspace tiers.

Reviewer 2: ```suggestion
* Available to all non-Sandbox site plans, or sites in Gold or above workspace tiers.
```

* Sites must be updated to the [latest release](https://github.com/pantheon-systems/drops-7/tags) of Drupal core.
<br/>
<br/>

To verify your current Drupal core version, you can check the [Status report tab](/drupal-launch-check#status-report) in the Pantheon site dashboard or you can use Terminus: 

```bash{promptUser: user}
terminus drush <site>.<env> status
```

To verify your current upstream, you can go to the Pantheon site dashboard and click **Settings**, then **About Site** or you can use Terminus: 

```bash{promptUser: user}
terminus site:info <site>
```

For sites not already using the Pantheon Drupal 7 Upstream (or a Custom Upstream based on the Pantheon Drupal 7 Upstream), we recommend hiring experts like [Tag1](https://www.tag1consulting.com/) to handle converting upstreams, which can be complex and cause disruption if done incorrectly.

#### D7 Contrib Modules LTS
This includes security patches and compatibility updates to community-contributed modules via Tag1’s [D7 Extended Support (D7ES) module](https://www.drupal.org/project/tag1_d7es), which will be distributed and maintained as part of the Pantheon Drupal 7 Upstream.

##### Requirements

> * Available to all site plans for sites associated with a Gold, Diamond, or Platinum [workspace account tier](/guides/account-mgmt/plans/workspace-plans).

Reviewer 2: ```suggestion
* Available to all non-Sandbox site plans, or sites associated with a Gold, Diamond, or Platinum [workspace account tier](/guides/account-mgmt/plans/workspace-plans).
```

* Sites must use the [Pantheon Drupal 7 Upstream](https://github.com/pantheon-systems/drops-7).
  * Or a Custom Upstream based on the Pantheon Drupal 7 Upstream with the [latest updates pulled in](/guides/custom-upstream/create-custom-upstream#pull-in-core-from-pantheons-upstream).
* Sites must be updated to the [latest release](https://github.com/pantheon-systems/drops-7/tags) of Drupal core.
* (Recommended) Not a requirement, but we strongly recommend updating all contrib modules to their latest versions as well. 

##### D7ES Usage
<Alert title="Note" type="info" >

This usage section is not yet applicable as the D7ES module has not been added to the Pantheon Drupal 7 Upstream.

However, you can expect the functionality and behavior described below to become available in the near future, before Drupal 7's end of life date.
</Alert>

> By enabling this module, your site’s list of contrib modules will be automatically submitted for D7ES security coverage. You will receive email notifications of D7 core & installed contrib module security and compatibility updates. 

Reviewer 2: ```suggestion
By enabling this module and configuring the opt-in, your site’s list of contrib modules will be automatically submitted for D7ES security coverage. You will receive email notifications of D7 core & installed contrib module security and compatibility updates. 
```
